aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -159,6 +159,12 @@ impl VMChangeSet {
             .chain(self.aggregator_write_set.iter())
     }
 
+    pub fn num_write_ops(&self) -> usize {
+        self.resource_write_set.len()
+            + self.module_write_set().len()
+            + self.aggregator_v1_write_set().len()
```

Comment:
shouldn't we be adding delta set as well, those will become writes too (not that it matters for V1, but logically seems true?)

Reply:
Oh.. actually this is after everything is materialized, there shouldn't be deltas.

Reply:
Are you sure? we could assert in that case, but I thought checker happens in the session and then it's likely it wouldn't be materialized @georgemitenkov 

delta gas is for sure smt we need to do :-)

Reply:
this can't be after materialization because then you wouldn't be able to abort the transaction?

Reply:
K I'm wrong. It's after user session before epilogue.
https://github.com/aptos-labs/aptos-core/blob/48aac6d3d9f654198defec4b4b0eeb034abd40e1/aptos-move/aptos-vm/src/move_vm_ext/session.rs#L385

Reply:
We should have a limit on deltas probably, but I think we can do that as a part of parallel execution gas (also behaviour between V1 and V2 is a bit different)? @gelash